pkgs/servers/nosql/mongodb/mongodb.nix

```diff
@@ -121,14 +121,14 @@ in stdenv.mkDerivation rec {
     # fix environment variable reading
     substituteInPlace SConstruct \
         --replace "env = Environment(" "env = Environment(ENV = os.environ,"
-   '' + lib.optionalString (versionAtLeast version "4.4" && versionOlder version "4.6") ''
+   '' + lib.optionalString (versionAtLeast version "4.4") ''
     # Fix debug gcc 11 and clang 12 builds on Fedora
     # https://github.com/mongodb/mongo/commit/e78b2bf6eaa0c43bd76dbb841add167b443d2bb0.patch
     substituteInPlace src/mongo/db/query/plan_summary_stats.h --replace '#include <string>' '#include <optional>
     #include <string>'
     substituteInPlace src/mongo/db/exec/plan_stats.h --replace '#include <string>' '#include <optional>
     #include <string>'
-  '' + lib.optionalString stdenv.isDarwin ''
+  '' + lib.optionalString (stdenv.isDarwin && versionOlder version "6.0") ''
```

Comment:
Did we break something patching just for 6.x?

Reply:
Don't think so. This is basically saying that starting with mongoDB v6 that `substituteInPlace` statement is not needed anymore. So versions <6.0 should continue to work as they will receive the patch.